### PR TITLE
🌱 Controller-runtime v0.19.0 & Controller-tools v0.16.1

### PR DESCRIPTION
**Pull request by akutz**
**What does this PR do, and why is it needed?**

This patch updates VM Operator to Controller-Runtime v0.19.0 and Controller-Tools v0.16.1.

Please note that this change apparently requires updating to controller-tools [v0.16](https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.16.0) as well, or else manifest generation fails per https://github.com/vmware-tanzu/vm-operator/pull/675#issuecomment-2299571871. To the VM Op team, please note one aspect of controller-tools v0.16 is deduplicating RBAC per https://github.com/kubernetes-sigs/controller-tools/pull/937, hence the major diff to our `role.yaml` file.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes `NA`

**Are there any special notes for your reviewer**:

cc @sbueringer 

**Please add a release note if necessary**:

```release-note
Update Controller-Runtime to v0.19.0 and Controller-Tools v0.16.1.
```

### CODE REVIEW

**Comment by akutz** [comment]
@sbueringer , it's failing to run controller-tools. It works once I upgrade it to v0.16. Does controller-runtime v0.19.0 somehow force a dependency on controller-tools v0.16+?

**Review by aruneshpa — APPROVED** [review]
I tried to cross references a bunch of randomly references roles to make sure they were ported correctly.  Seems so.

LGTM.

**Comment by sbueringer** [comment]
@akutz 

> It works once I upgrade controller-tools to v0.16. Does controller-runtime v0.19.0 somehow force a dependency on controller-tools v0.16+?

It's not controller-runtime that does it, it's rather k8s.io/api v0.31.0 (xref: https://github.com/kubernetes-sigs/controller-tools/pull/1000)

**Comment by sbueringer** [comment]
lgtm +/- the linter findings

**Comment by sbueringer** [comment]
> I tried to cross references a bunch of randomly references roles to make sure they were ported correctly. Seems so.

We diff'ed all our roles in Cluster API when making the change in controller-gen and we have a lot :) (just for some additional confidence)

File: pkg/prober/worker/prober_worker.go [hunk 1]
```diff
@@ -12,9 +12,11 @@ import (
 	"github.com/vmware-tanzu/vm-operator/pkg/prober/probe"
 )
 
+type DelayingInterface = workqueue.TypedDelayingInterface[any]
```
Comment (akutz) [inline pkg/prober/worker/prober_worker.go:15]: @sbueringer I'll admit, I'm punting this one down the road a bit 😃

**Pull request merged**